client/src/components/Contacts.jsx

```diff
@@ -0,0 +1,62 @@
+import Box from "@mui/material/Box";
+import TableContainer from "@mui/material/TableContainer";
+import { useEffect, useState } from "react";
+import { fetchContacts } from "../fetchContacts";
+import DeleteConfirmation from "./DeleteConfirmation";
+import ContactTable from "./ContactTable";
+
+const Contacts = ({ setSelectedContactId, setContacts, contacts }) => {
+  const [deleteConfirmationOpen, setDeleteConfirmationOpen] = useState(false);
+  const [contactToDelete, setContactToDelete] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const contactsData = await fetchContacts();
+      setContacts(contactsData);
+    };
+    fetchData();
```

Comment:
We should always try to avoid too generally vague names for variables and functions. `fetchData()` is such a vague name. In this case, it's pretty obvious what it does - it's simply an async wrapper function for calling `fetchContacts()` inside a `useEffect()`, which doesn't allow directly calling async code within its callback function. Maybe we can rename this to be "fetchContactsOnPageLoad()" or "loadPageContacts()" something like that?